CMake/genex.cmake

```diff
@@ -0,0 +1,28 @@
+# Generator expression helpers
+
+# Provide an option that defaults to ON in debug builds.
+macro(DEBUG_OPTION name description)
+  option(${name} ${description} FOR_DEBUG)
+endmacro()
+
+# Provide an option that defaults to ON in non-debug builds.
+macro(NON_DEBUG_OPTION name description)

```

Comment:
Maybe a better name would be `RELEASE_OPTION`

Reply:
Done, along with some other improvements